File: 2020-week12/c++/2020-week12_1.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes)
{

    // cloth count
    map<string, int> m;
    for (int i = 0; i < clothes.size(); i++)
    {
        m[clothes[i][1]]++;
    }

    vector<int> list;
    for (auto iter = m.begin(); iter != m.end(); iter++)
    {
        list.push_back(iter->second);
    }

    // 모든 경우의 수
    int answer = 1;
    for (int x : list)
    {
        answer *= (x + 1);
    }
    return answer - 1;
}

File: 2020-week12/js/2020-week12_1.js
function solution(clothes) {
  var answer = 1;

  let closet = {};
  clothes.forEach((element) => {
    if (closet.hasOwnProperty(element[1])) {
      closet[element[1]].push(element[0]);
    } else {
      closet[element[1]] = [element[0]];
    }
  });

  const keys = Object.keys(closet);

  if (keys.length === 1) answer = closet[clothes[0][1]].length;
  else {
    keys.forEach((key) => {
      answer *= closet[key].length + 1;
    });
    answer -= 1;
  }

  return answer;
}
